### PR TITLE
falling back to the release definition name if definitionId is null

### PR DESCRIPTION
**Pull request by ankitgo**

### CODE REVIEW

**Comment by Dan-Co** [comment]
Great thanks, ill give this version a run tomorrow and report back.

File: src/Agent.Worker/Release/ReleaseJobExtension.cs [hunk 1]
```diff
@@ -268,11 +268,19 @@ private void InitializeAgent(IExecutionContext executionContext, out bool skipAr
...
             executionContext.Output($"ReleaseId={releaseId}, TeamProjectId={teamProjectId}, ReleaseDefinitionName={releaseDefinitionName}");
 
+            var releaseDefinition = executionContext.Variables.Get(Constants.Variables.Release.ReleaseDefinitionId);
+            if (string.IsNullOrEmpty(releaseDefinition))
+            {
+                string pattern = new string(Path.GetInvalidFileNameChars()) + new string(Path.GetInvalidPathChars());
+                Regex regex = new Regex(string.Format("[{0}]", Regex.Escape(pattern)));
+                releaseDefinition = regex.Replace(releaseDefinitionName, string.Empty);
+            }
+
```
Comment (GitHubSriramB) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: let's trace the path we construct

**Comment by ankitgo** [comment]
@Dan-Co thanks. let me know if this fix works for you.

**Comment by Dan-Co** [comment]
Hi,
Just realised I cannot actually build the solution locally because we have corporate lock down on our pcs (cannot install dot.net.core etc..)
Is there a pre-built version I can get?

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 1 above)
Comment (ankitgo) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: Added this to another PR since its very minor

**Comment by ankitgo** [comment]
@Dan-Co drop me an email at angoya@microsoft.com . We ll take it forward from there.

File: src/Agent.Worker/Release/ReleaseJobExtension.cs (hunk 1 above)
Comment (ankitgo) [inline src/Agent.Worker/Release/ReleaseJobExtension.cs]: Yes we are logging the mapping file.

**Comment by kasubram** [comment]
looks good

**Comment by Dan-Co** [comment]
Ahh, didnt get this on work email.

Ill wait for the build to occur against this commit, I guess its imminent?

**Pull request merged**